src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1124,20 +1124,97 @@ protected function xmlToProps($xml)
     protected function xmlToColumns($xml, $propertyNames)
     {
         $data = new stdClass();
-        $dom = new DOMDocument('1.0');
-        $dom->loadXML($xml);
-        $dom->formatOutput = true;
-        $xpath = new DOMXPath($dom);
 
-        foreach ($propertyNames as $propertyName) {
-            $els = $xpath->query(sprintf('.//sv:property[@sv:name="%s"]', $propertyName));
-            if ($els->length === 0) {
-                continue;
+        $lastPropertyName = null;
```

Comment:
this could be moved into a class then the callback method can access this properties with `$this->lastPropertyName` instead of using `&` must look into it what performance impact that will have.

Reply:
i like that idea, also because the Client is already huge and adding this inside a method does not improve readability ;-)